src/vma/dev/qp_mgr.cpp

```diff
@@ -587,8 +592,13 @@ void qp_mgr::mlx5_send(vma_ibv_send_wr *p_send_wqe)
 	 * implementations may use move-string-buffer assembler instructions,
 	 * which do not guarantee order of copying.
 	 */
+	if (likely(!m_b_is_hypervisor)) {
```

Comment:
Must we have an if statement on the critical path? Please note the following link: https://gcc.gnu.org/onlinedocs/gcc-4.9.0/gcc/Function-Multiversioning.html 
Can you use Function Multi Versioning for SSE3 targets?